File: frontend/tests/utils/mock/types/detector.ts
import type { DetectorConfigurationNode, DetectorNode } from '../../../../src/api/types.gql-generated';

export type DetectorConfiguration =
  Omit<DetectorConfigurationNode, 'detector' | 'annotations'>
export type Detector =
  Omit<DetectorNode, 'specification' | 'configurations'>

export const detectorConfiguration: DetectorConfiguration = {
  id: '1',
  configuration: 'Test configuration',
}
export const detector: Detector = {
  id: '1',
  name: 'detector1',
}

File: backend/api/schema/nodes/detector.py
import graphene
import graphene_django_optimizer

from backend.api.models import Detector
from backend.utils.schema.types import BaseObjectType, BaseNode
from .detector_configuration import DetectorConfigurationNode


class DetectorNode(BaseObjectType):
    """Detector schema"""

    class Meta:
        model = Detector
        fields = "__all__"
        filter_fields = {}
        interfaces = (BaseNode,)

    configurations = graphene.List(DetectorConfigurationNode)

    @graphene_django_optimizer.resolver_hints()
    def resolve_configurations(self: Detector, info):
        return self.configurations.all()
